### PR TITLE
[Incremental] Add a swiftdeps deserialization performance test

### PR DESCRIPTION
**Pull request by davidungar**
Adds a test that reads and integrates swiftdeps files in order to optimize the problematic phase of clean builds. Run it using instruments with the scheme set to use optimized code in order to get a good profile.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
In service of rdar://79192750 , rdar://80094226

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test

**Review by artemcm — COMMENTED** [review]
Thanks for adding this test, @davidungar!

Thinking about maintenance of this test, we have an additional burden of ensuring that the `.swiftdeps` inputs we are adding here have the serialization format version that exactly matches that of the driver. If we are bumping this version in the future across the compiler and the driver, the test will start failing until the inputs are re-generated with the then-new serialization format. 

To ease the maintenance, it would really help to:
1. Provide some sort of automation to re-generate these inputs and put them in the right place. (kind of like we have to do for `Options` today).
2. Warn developers updating the serialization version numbers in the code that this automation must also run for this test to pass.

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@artemcm I've added directions, not quite automation. What do you think? Good enough??

File: Tests/SwiftDriverTests/CleanBuildPerformanceTests.swift [hunk 1]
```diff
@@ -6,13 +6,25 @@ import XCTest
...
   /// When doing an incremental but clean build, after every file is compiled, its `swiftdeps` file must be
   /// deserialized and integrated into the `ModuleDependencyGraph`.
   /// This test allows us to profile an optimize this work. (Set up the scheme to run optimized code.)
   /// It reads and integrages every swiftdeps file in a given directory.
+  ///
+  /// This test relies on sample `swiftdeps` files to be present in `<project-folder>/TestInputs/SampleSwiftDeps`.
+  /// If the serialization format changes, they will need to be regenerated.
+  /// To regenerate them:
+  /// - Remove the existing files in `<project-folder>/TestInputs/SampleSwiftDeps`
+  /// - Do a clean, debug build of this project.
```
Comment (artemcm) [inline Tests/SwiftDriverTests/CleanBuildPerformanceTests.swift:21]: We do not document a way of building this package as an Xcode project in the README (FWIW, I've never done so either).
Furthermore, SwiftPM, for the `swift package generate-xcodeproj` command, currently emits:
```
'generate-xcodeproj' is no longer needed and will be deprecated soon.
```
Could we please generalize these instructions for building with SwiftPM? That way non-Darwin users also have a way of following these steps.

Comment (artemcm) [inline Tests/SwiftDriverTests/CleanBuildPerformanceTests.swift:21]: Is this using `swift package generate-xcodeproj`?
We do not document a way of building this package as an Xcode project in the README (FWIW, I've never done so either). Furthermore, SwiftPM, for the `swift package generate-xcodeproj` command, currently emits:
```
'generate-xcodeproj' is no longer needed and will be deprecated soon.
```
Could we please generalize these instructions for building with SwiftPM? That way non-Darwin users also have a way of following these steps.

Comment (davidungar) [reply Tests/SwiftDriverTests/CleanBuildPerformanceTests.swift]: Thanks a lot for such a prompt review! Thanks to you, I see the ambiguity. I've come up with something that should be better.

**Review by artemcm — APPROVED** [review]

**Comment by artemcm** [comment]
Thank you!

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
> Thank you!

My pleasure! Times like these, I really appreciate the utility of the review process. (BTW, I added one more wrinkle to the shell script, to clear out the old build directory.)

**Pull request merged**